firefox-ios/Client/Coordinators/Library/LibraryCoordinator.swift

```diff
@@ -135,12 +135,55 @@ class LibraryCoordinator: BaseCoordinator,
         if isBookmarkRefactorEnabled {
             (navigationController.topViewController as? BookmarksViewController)?
                 .bookmarkCoordinatorDelegate = bookmarksCoordinator
+            if let recentFolderGUID = profile.prefs.stringForKey(PrefsKeys.LastViewedBookmarkFolder) {
+                CATransaction.begin()
+                CATransaction.setDisableActions(true)
+
+                populateBookmarksNavigationController(currentFolderGUID: recentFolderGUID,
+                                                      bookmarksCoordinator: bookmarksCoordinator) {
+                        CATransaction.commit()
+                }
+            }
         } else {
             (navigationController.topViewController as? LegacyBookmarksPanel)?
                 .bookmarkCoordinatorDelegate = bookmarksCoordinator
         }
     }
 
+    private func populateBookmarksNavigationController(currentFolderGUID: String,
+                                                       bookmarksCoordinator: BookmarksCoordinator,
+                                                       completion: @escaping () -> Void) {
+        profile.places.getBookmarksTree(rootGUID: BookmarkRoots.MobileFolderGUID,
```

Comment:
I am not sure anymore, is this a network call, or is this persisted in the rust layer? Wondering if we should worry about a slow internet connection here or not. If so, can you test what would happen visually if we have a very slow internet connection?

Reply:
Good call, I can try to test that 👍

Reply:
Just tried throttling my internet speed (and even turning off my internet)
<img width="386" alt="Screenshot 2025-01-09 at 9 37 42 AM" src="https://github.com/user-attachments/assets/bd752520-e991-4763-96f7-fde2a096eb99" />
and opening a bookmarks folder in a way that would call this method, it opened without any delay and with the proper items in the folder. I think this would indicate some sort of caching/persistence that would circumvent any network delay?

https://github.com/user-attachments/assets/8659b0ca-ddb7-419d-a2db-7c8e57d43490


https://github.com/user-attachments/assets/60b20af1-8726-446a-86e8-f79779cc540f

Reply:
Awesome thank you for having a look!